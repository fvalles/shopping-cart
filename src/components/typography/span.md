Basic span text component with default color="productPrice":

```jsx
<Span>Lorem ipsum</Span>
```

Span text component with default color="productPrice" and margin="0px 0px 0px 4px":

```jsx
<Span margin="0px 0px 0px 4px">Lorem ipsum</Span>
```

Span text component with color="primary":

```jsx
<Span color="primary">Lorem ipsum</Span>
```
