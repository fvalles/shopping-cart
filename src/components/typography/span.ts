import styled, { css } from 'styled-components';
import { TypographyProps } from './common';

export const Span = styled.span<TypographyProps>`
  color: ${({ color = 'productPrice', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.regular}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 16px;
  line-height: 1;
  ${({ margin }) =>
    margin &&
    css`
      margin: ${margin};
    `}
`;
