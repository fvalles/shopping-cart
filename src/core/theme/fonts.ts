export const Fonts = {
  light: 'Avenir Light',
  medium: 'Avenir Medium',
  regular: 'Avenir Regular',
};

export type FontsType = typeof Fonts;
