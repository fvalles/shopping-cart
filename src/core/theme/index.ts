export { Colors } from './colors';
export { Fonts } from './fonts';
export { GlobalStyle } from './global-style';
