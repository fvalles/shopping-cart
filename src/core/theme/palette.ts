export const palette = {
  // color names are based on https://colornamer.netlify.app/
  black: '#000000',
  blackMarket: '#212240',
  blackMarketLight: '#21224029',
  bleachedSilk: '#F3F3F3',
  faintFern: '#DBDBE0',
  mischka: '#A6A7B3',
  sharpGrey: '#CACAD1',
  stormGrey: '#717285',
  venetianNights: '#7350FF',
  white: '#FFFFFF',
};
