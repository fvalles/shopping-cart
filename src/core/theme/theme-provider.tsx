import React, { FunctionComponent } from 'react';
import { DefaultTheme, ThemeProvider as Provider } from 'styled-components';
import { Colors, Fonts } from '.';

interface ThemeProviderProps {
  children?: React.ReactNode;
}

export const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  children,
}) => (
  <Provider
    theme={
      {
        Colors,
        Fonts,
      } as DefaultTheme
    }>
    {children}
  </Provider>
);
