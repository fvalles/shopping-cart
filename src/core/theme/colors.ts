import { palette } from './palette';

export const Colors = {
  // color names are based on https://colornamer.netlify.app/
  primary: palette.venetianNights,
  productImageBorder: palette.sharpGrey,
  productPrice: palette.black,
  productQuantity: palette.faintFern,
  productTitle: palette.mischka,
  summaryBackground: palette.bleachedSilk,
  summaryHeading: palette.stormGrey,
  summaryText: palette.blackMarket,
  white: palette.white,
};

export type ColorsType = typeof Colors;
export type KeyColors = keyof ColorsType;
