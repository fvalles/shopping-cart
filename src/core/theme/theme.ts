import { DefaultTheme } from 'styled-components';
import { Colors } from './colors';

const theme: DefaultTheme = {
  Colors,
};

export { theme };
