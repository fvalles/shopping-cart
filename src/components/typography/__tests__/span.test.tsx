import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Span } from '../span';

describe('span component', () => {
  it('should render with default css properties', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Span>Lorem ipsum</Span>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('color', '#000000');
    expect(result).toHaveStyleRule(
      'font-family',
      "Avenir Regular,'Avenir Next','Segoe UI', sans-serif",
    );
    expect(result).toHaveStyleRule('font-size', '16px');
    expect(result).toHaveStyleRule('line-height', '1');
    expect(result).toMatchSnapshot();
  });

  it('should render with custom white color and 20px margin', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Span color="white" margin="20px">
            Lorem ipsum
          </Span>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('color', '#FFFFFF');
    expect(result).toHaveStyleRule(
      'font-family',
      "Avenir Regular,'Avenir Next','Segoe UI', sans-serif",
    );
    expect(result).toHaveStyleRule('font-size', '16px');
    expect(result).toHaveStyleRule('line-height', '1');
    expect(result).toHaveStyleRule('margin', '20px');
    expect(result).toMatchSnapshot();
  });
});
