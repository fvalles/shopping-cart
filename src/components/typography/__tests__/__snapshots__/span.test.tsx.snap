// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`span component should render with custom white color and 20px margin 1`] = `
.c0 {
  color: #FFFFFF;
  font-family: Avenir Regular,'Avenir Next','Segoe UI', sans-serif;
  font-size: 16px;
  line-height: 1;
  margin: 20px;
}

<span
  className="c0"
  color="white"
>
  Lorem ipsum
</span>
`;

exports[`span component should render with default css properties 1`] = `
.c0 {
  color: #000000;
  font-family: Avenir Regular,'Avenir Next','Segoe UI', sans-serif;
  font-size: 16px;
  line-height: 1;
}

<span
  className="c0"
>
  Lorem ipsum
</span>
`;
